ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py

```diff
@@ -1,10 +1,12 @@
 import ivy
+from ivy.functional.frontends.numpy.func_wrapper import to_ivy_arrays_and_back
 
 
+@to_ivy_arrays_and_back
 def diagonal(x, offset, axis1, axis2):
-    ivy.assertions.check_equal(
-        axis1 == axis2, message="Both axis values should not be the same"
-    )
-    ivy.assertions.check_equal(x.ndim, 2, message="x must be 2-dimensional")
+    # ivy.assertions.check_equal(
+    #   axis1 == axis2, message="Both axis values should not be the same"
+    # )
+    # ivy.assertions.check_equal(x.ndim, 2, message="x must be 2-dimensional")
 
-    return ivy.diagonal(x, offset=offset, axis1=axis1, axis2=axis2)
```

Comment:
This is the function you are calling https://github.com/unifyai/ivy/blob/f7a2242c59faaeb29d42482f768d3c46618077b9/ivy/functional/ivy/linear_algebra.py#L326
Only the first argument `x` is positional. The rest of the arguments should be keyword.